omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -960,22 +783,36 @@ def _ik_solver_cartesian_to_joint_space(self, relative_target_pose):
                 - th.tensor or None: Joint positions to reach target pose or None if impossible to reach the target pose
                 - th.tensor: Indices for joints in the robot
         """
+        if arm is None:
+            arm = self.arm
+        # TODO: make sure descriptor file match torso & arm setting
         ik_solver = IKSolver(
-            robot_description_path=self._manipulation_descriptor_path,
+            robot_description_path=self.robot.robot_arm_descriptor_yamls[arm],
             robot_urdf_path=self.robot.urdf_path,
-            reset_joint_pos=self.robot.reset_joint_pos[self._manipulation_control_idx],
+            reset_joint_pos=self.robot.reset_joint_pos[self._manipulation_control_idx(arm)],
```

Comment:
oh `self._manipulation_control_idx` is used here - so if we get rid of lula completely, we don't need this any more?